### DISCUSSION TITLE
How do I set a backup font in case the set font doesn't include a certain Unicode character?

### DISCUSSION THREAD

**Discussion by xuhdev** [syl20bnr/spacemacs#16693] [Q&A]
I currently use Inconsolata as the font:

```elisp
   dotspacemacs-default-font '("Inconsolata"
                               :size 15.5
                               :weight normal
                               :width normal)
```

However, Inconsolata (like Source Code Pro) doesn't include some Unicode characters (e.g., `→`, `←`) and it seems to fall back to something not monospaced. How can I set up a backup font for those characters?

**Comment by fnussbaum** [comment]
On my system, Inconsolata includes both arrows → and ←. I guess customizing `face-font-family-alternatives` to include an entry `("Inconsolata" "Monospace")` might work. But I would be interested to know other specific characters where this problem occurs such that I can try and reproduce it. Setting fonts for certain character sets should also be possible, see for example [this blog post](https://idiocy.org/emacs-fonts-and-fontsets.html).

**Reply by xuhdev** [reply]
@fnussbaum Turned out I already have "Monospace" in `face-font-family-alternatives`.

It also turns out my `Inconsolata` also includes `→`, but Emacs displays it non-monospaced. Do you see a similar effects in your Emacs?

![image](https://github.com/user-attachments/assets/2de676e2-a5b9-4317-9080-4b12bc750152)

**Comment by fnussbaum** [comment]
For me, it looks like this 
![image](https://github.com/user-attachments/assets/0a41ada5-04fd-4c18-9994-dded141414d9)
(which correctly uses the monospaced Inconsolata arrows). I'm not sure why the arrows use a different font on your system. I get a different non-monospaced font only for long arrows ⟶.

**Reply by xuhdev** [reply]
Not sure why, I eventually fixed this by installing the `Inconsolata` font from [Nerd Font](https://www.nerdfonts.com/font-downloads). The font name must also be changed to `Inconsolata Nerd Font Mono`.
